fe/fe-core/src/main/java/com/starrocks/lake/vacuum/AutovacuumDaemon.java

```diff
@@ -152,17 +155,19 @@ private void vacuumPartition(Database db, OlapTable table, PhysicalPartition par
     }
 
     private void vacuumPartitionImpl(Database db, OlapTable table, PhysicalPartition partition) {
-        List<Tablet> tablets;
+        Map<Long, List<Tablet>> indexToTablets = new HashMap<>();
         long visibleVersion;
         long minRetainVersion;
         long startTime = System.currentTimeMillis();
         long minActiveTxnId = computeMinActiveTxnId(db, table);
-        Map<ComputeNode, List<Long>> nodeToTablets = new HashMap<>();
+        Map<ComputeNode, List<TabletInfoPB>> nodeToTablets = new HashMap<>();
 
         Locker locker = new Locker();
         locker.lockTablesWithIntensiveDbLock(db.getId(), Lists.newArrayList(table.getId()), LockType.READ);
         try {
-            tablets = partition.getBaseIndex().getTablets();
+            for (MaterializedIndex index : partition.getMaterializedIndices(IndexExtState.VISIBLE)) {
+                indexToTablets.put(index.getId(), index.getTablets());
```

Comment:
`index.getTablets()` returns the reference of the tablets, manipulating the lists out of the lock may not be safe and could yield unexpected exception or inconsistent result.